### PR TITLE
Syncing license data

### PR DESCRIPTION
**Pull request by salahalsaleh1993**

### CODE REVIEW

File: pkg/pull/signature.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func verify(message, signature, publicKeyPEM []byte) error {
 
 func getMessageFromLicense(license *kotsv1beta1.License) ([]byte, error) {
 	// JSON marshaller will sort map keys automatically.
-	fields := map[string]string{
+	fields := map[string]interface{}{
```
Comment (divolgin) [inline pkg/pull/signature.go]: I think changing this to interface opens this up to programming mistakes.  You can now add an object of any type to this map.

Comment (salahalsaleh1993) [reply pkg/pull/signature.go]: the license is signed with `licenseSequence` being an integer, that was the reason for changing this

File: ffi/main.go [hunk 2]
```diff
@@ -133,6 +137,87 @@ func UpdateCheck(socket, fromArchivePath string) {
...
+
+	u, err := url.Parse(license.Spec.Endpoint)
+	if err != nil {
+		fmt.Printf("failed to parse endpoint from license: %s\n", err.Error())
+		return C.CString("")
+	}
+
+	hostname := u.Hostname()
+	if u.Port() != "" {
+		hostname = fmt.Sprintf("%s:%s", u.Hostname(), u.Port())
```
Comment (divolgin) [inline ffi/main.go]: `net.JoinHostPort` > `fmt.Sprintf`

File: pkg/pull/signature.go (hunk 1 above)
Comment (salahalsaleh1993) [reply pkg/pull/signature.go]: i guess i should change signing the license to use `licenseSequence` as string

File: ffi/main.go [hunk 3]
```diff
@@ -133,6 +137,87 @@ func UpdateCheck(socket, fromArchivePath string) {
...
+	}
+
+	url := fmt.Sprintf("%s://%s/release/%s/license", u.Scheme, hostname, license.Spec.AppSlug)
```
Comment (divolgin) [inline ffi/main.go]: actually, you can just assume license endpoint format is what should be used, so you don't need to generate `hostname`

**Pull request merged**